app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt

```diff
@@ -1,207 +1,251 @@
+                        )
+                    }
+                }
 
-            stroboscopeBtn.setOnClickListener {
-                toggleStroboscope(false)
+                MainScreen(
+                    flashlightButton = {
+                        val flashlightActive by viewModel.flashlightOn.collectAsStateWithLifecycle()
+
+                        FlashlightButton(
+                            onFlashlightPress = { viewModel.toggleFlashlight() },
+                            flashlightActive = flashlightActive,
+                        )
+                    },
+                    brightDisplayButton = {
+                        val showBrightDisplayButton by preferences.brightDisplayFlow.collectAsStateWithLifecycle(
+                            config.brightDisplay,
+                            minActiveState = Lifecycle.State.CREATED
+                        )
+                        if (showBrightDisplayButton) {
+                            BrightDisplayButton(
+                                onBrightDisplayPress = {
+                                    startActivity(Intent(applicationContext, BrightDisplayActivity::class.java))
+                                }
+                            )
+                        }
+                    },
+                    sosButton = {
+                        val showSosButton by preferences.sosFlow.collectAsStateWithLifecycle(config.sos, minActiveState = Lifecycle.State.CREATED)
+                        val sosActive by viewModel.sosActive.collectAsStateWithLifecycle()
+
+                        if (showSosButton) {
+                            SosButton(
+                                sosActive = sosActive,
+                                onSosButtonPress = {
+                                    toggleStroboscope(true, sosPermissionLauncher)
+                                },
+                            )
+                        }
+                    },
+                    stroboscopeButton = {
+                        val showStroboscopeButton by preferences.stroboscopeFlow.collectAsStateWithLifecycle(
+                            config.stroboscope,
+                            minActiveState = Lifecycle.State.CREATED
+                        )
+                        val stroboscopeActive by viewModel.stroboscopeActive.collectAsStateWithLifecycle()
+
+                        if (showStroboscopeButton) {
+                            StroboscopeButton(
+                                stroboscopeActive = stroboscopeActive,
+                                onStroboscopeButtonPress = {
+                                    toggleStroboscope(false, stroboscopePermissionLauncher)
+                                },
+                            )
+                        }
+                    },
+                    slidersSection = {
+                        val brightnessBarVisible by viewModel.brightnessBarVisible.collectAsStateWithLifecycle(false)
```

Comment:
you can use `.stateIn` inside the view model to avoid passing a default argument here in the Composable, it shouldn't drive the logic.

Reply:
Also 2 lines below